lucet-runtime/lucet-runtime-internals/src/future.rs

```diff
@@ -101,51 +117,20 @@ impl InstanceHandle {
     /// that use `Vmctx::block_on` and provides the trampoline that `.await`s those futures on
     /// behalf of the guest.
     ///
+    /// If `runtime_bound` is provided, it will also pause the Wasm execution and yield a future
+    /// that resumes it after (approximately) that many Wasm opcodes have executed.
```

Comment:
This is a bit unclear to me: if I'm reading correctly, this method is not returning a future, it's running a loop until the instance returns its final result, or an error occurs. The `runtime_bound` makes the `poll` method on the underlying, generated future return every time that many Wasm opcodes have been executed, right?

Sorry if this is splitting hairs, I just want to make sure I understand the intended behavior.

Reply:
Yes, that's right; we await on the `YieldNow` future, which results in the `async fn`'s lowering containing a state in which it has returned from `poll()` and will resume execution on the next `poll()`.